src/expr/src/scalar/func.rs

```diff
@@ -2694,11 +2676,11 @@ impl BinaryFunc {
             | AddTimeInterval
             | SubTimeInterval => input1_type,
 
-            DateTruncInterval => ScalarType::Interval.nullable(true),
-
             AddDateInterval | SubDateInterval | AddDateTime | DateBinTimestamp
             | DateTruncTimestamp => ScalarType::Timestamp.nullable(true),
```

Comment:
```
AddDateInterval | SubDateInterval | AddDateTime | DateBinTimestamp
| DateTruncInterval | DateTruncTimestamp => ScalarType::Timestamp.nullable(true),
```

Reply:
I need this separately, DateTruncInterval needs "Interval" not timestamp

Reply:
You're absolutely correct, sorry for the fuzz!